staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -258,3 +290,158 @@ func (c *CacheDelegator) ReadinessCheck() error {
 func (c *CacheDelegator) RequestWatchProgress(ctx context.Context) error {
 	return c.storage.RequestWatchProgress(ctx)
 }
+
+func (c *CacheDelegator) Stop() {
+	c.stopOnce.Do(func() {
+		close(c.stopCh)
+	})
+	c.wg.Wait()
+}
+
+func newConsistencyChecker(resourcePrefix string, newListFunc func() runtime.Object, cacher storeCache, etcd store, clock clock.Clock) *consistencyChecker {
+	return &consistencyChecker{
+		resourcePrefix: resourcePrefix,
+		newListFunc:    newListFunc,
+		cacher:         cacher,
+		etcd:           etcd,
+		clock:          clock,
+	}
+}
+
+type consistencyChecker struct {
+	resourcePrefix string
+	newListFunc    func() runtime.Object
+
+	cacher storeCache
+	etcd   store
+	clock  clock.Clock
+}
+
+type storeCache interface {
+	store
+	Ready() bool
+}
+
+type store interface {
+	GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error
+}
+
+func (c consistencyChecker) startChecking(stopCh <-chan struct{}) {
+	delay := ConsistencyCheckDelay + time.Duration(rand.Int63n(int64(ConsistencyCheckPeriod-ConsistencyCheckDelay)))
+	select {
```

Comment:
Why can't we start with check immediately?

Reply:
To prevent cachers from all resources listing from etcd immediately after cache is initialized. That would needlessly double number of requests during apiserver startup.

Reply:
Can we use `wait.PollUnit()` instead?

Reply:
Done